src/storage-blob-preview/azext_storage_blob_preview/_params.py

```diff
@@ -368,6 +368,26 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
             c.extra('snapshot', snapshot_type)
             c.extra('if_tags_match_condition', tags_condition_type)
 
+    with self.argument_context('storage blob service-properties delete-policy update') as c:
+        c.argument('enable', arg_type=get_enum_type(['true', 'false']), help='Enables/disables soft-delete.')
```

Comment:
Why not use `get_three_state_flag()` as `arg_type`?

Reply:
Yes, we could. Here is to align with previous design in track 1 SDK.
https://github.com/Azure/azure-cli/blob/701f1cd0f6f09486193dee977e51a3892cb4bbc0/src/azure-cli/azure/cli/command_modules/storage/_params.py#L647

Reply:
I just copied all interface definition from main repo and replace the SDK used.